client/web/src/tour/components/Tour/TourAgent.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react'
+
+import CheckCircleIcon from 'mdi-react/CheckCircleIcon'
+import ReactDOM from 'react-dom'
+import { useLocation } from 'react-router-dom'
+
+import { TelemetryProps } from '@sourcegraph/shared/src/telemetry/telemetryService'
+import { Icon } from '@sourcegraph/wildcard'
+
+import { GETTING_STARTED_TOUR_MARKER } from './TourInfo'
+import { TourTaskType, TourTaskStepType } from './types'
+import { parseURIMarkers } from './utils'
+
+import styles from './Tour.module.scss'
+
+interface TourAgentProps extends TelemetryProps {
+    tasks: TourTaskType[]
+    onStepComplete: (step: TourTaskStepType) => void
+}
+
+/**
+ * Component to track TourTaskStepType.completeAfterEvents and show info box for steps.
+ */
+export const TourAgent: React.FunctionComponent<TourAgentProps> = React.memo(
+    ({ tasks, telemetryService, onStepComplete }) => {
+        // Agent 1: Track completion
+        useEffect(() => {
+            const filteredSteps = tasks.flatMap(task => task.steps).filter(step => step.completeAfterEvents)
+            telemetryService?.addEventLogListener?.(eventName => {
```

Comment:
Let's make sure we clean up the event listener again otherwise they might compound over the duration of a session!

```suggestion
            return telemetryService?.addEventLogListener?.(eventName => {
```

Reply:
Thanks for catching it! Fixed.